### ISSUE TITLE
Support for forwarding services headers

### ISSUE THREAD

**Issue by TheLastProject** [k9mail/k-9#5757]
**Is your feature request related to a problem? Please describe.**
I use SimpleLogin, which helps me more easily create one email alias per service. Because of this, my emails all look like they come from email addresses like reply+iwehobhuhuoeahfewouhb@simplelogin.co instead of the forwarded address.

**Describe the solution you'd like**
SimpleLogin adds several headers to each email, allowing you to see the original sender:

https://github.com/simple-login/app/blob/ee0ed7d9ec822bcd5355c0d61930aa0f1ddccdb0/app/email/headers.py#L27-L30

For example:
```
X-SimpleLogin-Envelope-From: noreply@github.com
X-SimpleLogin-Envelope-To: my_alias@example.com
```

It would be great if K-9 Mail could use these headers to display the from and to address that the emails are forwarded from.

**Describe alternatives you've considered**
Another option would be perhaps to talk to the SimpleLogin app, but this seems more complex and also harder to extend to other forwarding services like AnonAddy.

**Additional context**
This should probably be something the user should enable manually (something like Email -> Forwarding Service -> SimpleLogin), to prevent malicious emails using such headers to spoof.

**Comment by jondaley** [comment]
I'll bet this would be trickier than you are imagining.

Adding the simple login headers to the list of headers checked would be relatively easy, but I just looked up simple login (I was curious about how they do it - I use a custom email address for every vendor I talk to, and it works really great with pine/alpine, which is my desktop client, and it automatically uses the To address in the From address when it matches a pattern, and I can manually edit the From address if I want to do something else, but my system doesn't work great in K9 without setting up an identity for each of my infinite addresses, so I only have created maybe 20 identities for the common ones and then wait until I'm on the desktop to reply to the rest).

But, I host my own email server, so that part is easy for me and my customers, but simple login requires you to respond using their custom From address and then they set/hide your real from address and that is the problem I think for K9.

I think what you want K9 to do is to show the real from and to headers in your client, but then when actually sending the email send to an @simplelogin.co address and that seems like it would be a bunch harder (and potentially non-intuitive sometimes).  How does it work in other email clients?

The best I could think of is to display:

From: "realaddress@example.com by way of" <abcdefg@simplelogin.com>

So then you can see the real address but K9 would know to send the email to simplelogin without further manipulation.

It does mess up the real name that the receiver then sees, e.g. Jon Daley <jon@example.com> there might be a way of escaping " and other important characters, but parsing every possible way of doing names and email addresses quickly gets out of hand.

**Comment by TheLastProject** [comment]
> How does it work in other email clients?

The same as with K-9 Mail currently:
```
From: Jon Daley - notifications at github.com <reply+abcdefghijklmnopqrstuvwxyz@simplelogin.co>
Subject: Re: [k9mail/k-9] Support for forwarding services headers (Issue #5757)
Reply-To: reply_abcdefghijklmnopqrstuvwxyz_abcdefg@simplelogin.co
To: k9mail/k-9 - k-9 at noreply.github.com <ra+k-9.at.noreply.github.com+abcdefg@simplelogin.co>
```

> From: "realaddress@example.com by way of" abcdefg@simplelogin.com

Oh, I didn't think of that. That could also fix the phishing risk

> So then you can see the real address but K9 would know to send the email to simplelogin without further manipulation.

Wouldn't K-9 already know where to send it? After all, the X-SimpeLogin-* headers are solely for how to display it in the client, not to change the email address that gets replied to. It should be a purely visual change.

**Comment by jondaley** [comment]
Right, reply to is easy. So that part is already handled by simplelogin and doesn't need any change?

I think you might be right, I was thinking you needed more than just a display change.  So maybe it is an easier request than I was thinking.

(The whole reason I'm here is that I love K9, and I saw a developer post that sounded like they might need some extra help, so I might take a look through some bug/feature requests to see what I can do)
